### PR TITLE
Feature/#16 should support docker provider

### PR DESCRIPTION
**Pull request by hieptranquoc**
connect #16 
please review this PR. Thank you!

### CODE REVIEW

**Comment by hieptranquoc** [comment]
@hoatle @datphan, please help me review this PR! thanks.

File: lib/teracy-dev-core.rb [hunk 1]
```diff
@@ -26,10 +25,8 @@ def self.init
     TeracyDev.register_configurator(TeracyDevCore::Config::SSH.new)
     TeracyDev.register_configurator(TeracyDevCore::Config::SyncedFolders.new)
     TeracyDev.register_configurator(TeracyDevCore::Config::Vgrant.new)
-    TeracyDev.register_configurator(TeracyDevCore::Config::VirtualBoxProvider.new)
-    TeracyDev.register_configurator(TeracyDevCore::Config::VMwareProvider.new) # Use for vmware provider.
+    TeracyDev.register_configurator(TeracyDevCore::Config::Provider.new)
     TeracyDev.register_configurator(TeracyDevCore::Config::WinRM.new)
     TeracyDev.register_configurator(TeracyDevCore::Config::WinSSH.new)
   end
```
Comment (hoatle) [inline lib/teracy-dev-core.rb]:
```
self.register_provider("docker", Providers::Docker.new)
```

**Comment by hoatle** [comment]
depends on https://github.com/teracyhq-incubator/teracy-dev-core/pull/41

**Review by hoatle — DISMISSED** [review]
pls check my comment on the associated issue to update this PR

**Comment by hieptranquoc** [comment]
I've updated! Please check again!

**Comment by hoatle** [comment]
@phuonglm pls review on this

File: lib/teracy-dev-core/providers/docker.rb [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+module TeracyDevCore
+  module Providers
+    # see: https://www.vagrantup.com/docs/docker/
+    class Docker < TeracyDevCore::Providers::Provider
+
+      def configure(provider_settings, node_config)
+        options = provider_settings.dup
+
+        ["_id", "type", "enabled"].each do |key|
```
Comment (hoatle) [inline lib/teracy-dev-core/providers/docker.rb]: move this common stuff to the provider configurator scope: "_id", "type", "enabled", "weight"

**Comment by hieptranquoc** [comment]
I've updated, please check PR again. thank you!

**Review by phuonglm — APPROVED** [review]

**Review by hoatle — APPROVED** [review]
LGTM

**Pull request merged**